crates/katana/core/src/messaging/starknet_messenger.rs

```diff
@@ -0,0 +1,361 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use anyhow::Result;
+use async_trait::async_trait;
+use starknet::accounts::{Account, Call, ExecutionEncoding, SingleOwnerAccount};
+use starknet::core::types::{BlockId, BlockTag, EmittedEvent, EventFilter, FieldElement, MsgToL1};
+use starknet::core::utils::starknet_keccak;
+use starknet::macros::{felt, selector};
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::{AnyProvider, JsonRpcClient, Provider};
+use starknet::signers::{LocalWallet, SigningKey};
+use starknet_api::core::{ContractAddress, EntryPointSelector, Nonce};
+use starknet_api::hash::{self, StarkFelt};
+use starknet_api::stark_felt;
+use starknet_api::transaction::{
+    Calldata, L1HandlerTransaction as ApiL1HandlerTransaction, TransactionHash, TransactionVersion,
+};
+use tokio::sync::RwLock as AsyncRwLock;
+use tracing::{debug, error, trace, warn};
+use url::Url;
+
+use crate::backend::storage::transaction::L1HandlerTransaction;
+use crate::messaging::{
+    MessagingConfig, Messenger, MessengerError, MessengerResult, MSGING_TARGET,
+};
+
+/// As messaging in starknet is only possible with EthAddress in the `to_address`
+/// field, we have to set magic value to understand what the user want to do.
+/// In the case of execution -> the felt 'EXE' will be passed.
+/// And for normal messages, the felt 'MSG' is used.
+/// Those values are very not likely a valid account address on starknet.
+const MSG_MAGIC: FieldElement = felt!("0x4d5347");
+const EXE_MAGIC: FieldElement = felt!("0x455845");
+
+pub const HASH_EXEC: FieldElement = felt!("0xee");
+
+///
+pub struct StarknetMessenger {
+    chain_id: FieldElement,
+    provider: AnyProvider,
+    wallet: LocalWallet,
+    sender_account_address: FieldElement,
+    messaging_contract_address: FieldElement,
+}
+
+impl StarknetMessenger {
+    pub async fn new(config: MessagingConfig) -> Result<Arc<AsyncRwLock<StarknetMessenger>>> {
```

Comment:
```suggestion
    pub async fn new(config: MessagingConfig) -> Result<StarknetMessenger> {
```
same reason as above for `EthereumMessenger`